### PR TITLE
726 - Complete settings documentation

### LINKED ISSUE #726: Document all wagtail settings
This ticket proposes to have all wagtail settings documented in a single location much like [Django does](https://docs.djangoproject.com/en/1.7/ref/settings/). Many of these settings are mentioned, if not fully documented, in howto/settings.rst but as you can see below some are not mentioned at all in the docs. 

howto/settings.rst is a general guide on "how to get your settings file configured for Django & wagtail" and includes a full example settings file as well as urls. I propose cleaning this document up and moving the documentation of the settings themselves to a dedicated page like Django has for all user overridable settings and then have the rest of the documentation link to this.

For reference, here is the list of all overridable settings and where they are currently referenced in the docs (if at all):
- WAGTAILFRONTENDCACHE
  - wagtail/docs/contrib_components/frontendcache.rst
- WAGTAILFRONTENDCACHE_LOCATION
- WAGTAILADMIN_NOTIFICATION_FROM_EMAIL
  - wagtail/docs/howto/settings.rst
- WAGTAIL_USAGE_COUNT_ENABLED
  - wagtail/docs/releases/0.5.rst
- WAGTAIL_PASSWORD_MANAGEMENT_ENABLED
- WAGTAIL_SITE_NAME
  - wagtail/docs/howto/settings.rst
- WAGTAILEMBEDS_EMBED_FINDER
  - wagtail/docs/howto/settings.rst
- WAGTAILIMAGES_IMAGE_MODEL
  - wagtail/docs/howto/settings.rst
  - wagtail/docs/releases/0.5.rst
- WAGTAILIMAGES_FEATURE_DETECTION_ENABLED
  - wagtail/docs/core_components/images/feature_detection.rst
  - wagtail/docs/getting_started/installation.rst
- WAGTAILSEARCH_BACKENDS
  - wagtail/docs/core_components/search/backends.rst
  - wagtail/docs/core_components/search/indexing.rst
  - wagtail/docs/getting_started/installation.rst
  - wagtail/docs/howto/settings.rst
  - wagtail/docs/reference/management_commands.rst
  - wagtail/docs/releases/0.7.rst
- WAGTAILSEARCH_RESULTS_TEMPLATE
  - wagtail/docs/core_components/search/searching.rst
  - wagtail/docs/howto/settings.rst
- WAGTAILSEARCH_RESULTS_TEMPLATE_AJAX
  - wagtail/docs/core_components/search/searching.rst
  - wagtail/docs/howto/settings.rst

### PR DESCRIPTION
**Pull request by lb-**
This PR should close #726

This original issue was probably created before we had the [Settings](http://docs.wagtail.io/en/v2.0.1/advanced_topics/settings.html) page.

Nonetheless there were still a few settings I found that were not documented on this page (or elsewhere).

Below is a checklist of the changes.

## Settings either in the original issue or found in the codebase

* [x] WAGTAIL_FRONTEND_LOGIN_TEMPLATE **found in code**
* [x] WAGTAIL_FRONTEND_LOGIN_URL **found in code**
* [x] WAGTAIL_PASSWORD_MANAGEMENT_ENABLED
* [x] WAGTAIL_PASSWORD_RESET_ENABLED **found in code**
* [x] WAGTAIL_SITE_NAME
* [x] WAGTAILAPI_BASE_URL **found in code**
* [x] WAGTAILAPI_LIMIT_MAX **found in code**
* [x] WAGTAILAPI_SEARCH_ENABLED **found in code**
* [x] WAGTAILAPI_USE_FRONTENDCACHE **found in code**
* [x] WAGTAILADMIN_NOTIFICATION_FROM_EMAIL
* [x] WAGTAILADMIN_RICH_TEXT_EDITORS **found in code**
* [x] WAGTAILEMBEDS_EMBED_FINDER **deprecated in 1.12 - not adding**
* [x] WAGTAILEMBEDS_EMBEDLY_KEY **deprecated in 1.12 - not adding**
* [x] WAGTAILFRONTENDCACHE
* [x] WAGTAILFRONTENDCACHE_LOCATION **no longer used - but added a note**
* [x] WAGTAILFRONTENDCACHE_LANGUAGES **added in 2.1**
* [x] WAGTAILIMAGES_BACKENDS
* [x] WAGTAILIMAGES_FEATURE_DETECTION_ENABLED
* [x] WAGTAILIMAGES_IMAGE_MODEL
* [x] WAGTAILSEARCH_BACKENDS
* [x] WAGTAILSEARCH_RESULTS_TEMPLATE
* [x] WAGTAILSEARCH_RESULTS_TEMPLATE_AJAX

## Settings added to the docs at `advanced_topics/settings.rst`

* WAGTAIL_FRONTEND_LOGIN_TEMPLATE
* WAGTAIL_FRONTEND_LOGIN_URL
* WAGTAILADMIN_RICH_TEXT_EDITORS (added for completeness but with note about hallo deprecation)
* WAGTAILAPI_BASE_URL
* WAGTAILAPI_LIMIT_MAX
* WAGTAILAPI_SEARCH_ENABLED
* WAGTAILAPI_USE_FRONTENDCACHE
* WAGTAILFRONTENDCACHE
* WAGTAILFRONTENDCACHE_LANGUAGES
* WAGTAILFRONTENDCACHE_LOCATION (just a note)
* WAGTAILIMAGES_BACKENDS
* WAGTAILIMAGES_FEATURE_DETECTION_ENABLED (just a note - deprecated)

## Settings already in the docs at `advanced_topics/settings.rst`

* TAG_SPACES_ALLOWED
* TAGGIT_CASE_INSENSITIVE
* WAGTAIL_ALLOW_UNICODE_SLUGS
* WAGTAIL_APPEND_SLASH
* WAGTAIL_AUTO_UPDATE_PREVIEW
* WAGTAIL_DATE_FORMAT
* WAGTAIL_DATETIME_FORMAT
* WAGTAIL_ENABLE_UPDATE_CHECK
* WAGTAIL_GRAVATAR_PROVIDER_URL
* WAGTAIL_PASSWORD_MANAGEMENT_ENABLED
* WAGTAIL_PASSWORD_RESET_ENABLED
* WAGTAIL_SITE_NAME
* WAGTAIL_USAGE_COUNT_ENABLED
* WAGTAIL_USER_CREATION_FORM
* WAGTAIL_USER_CUSTOM_FIELDS
* WAGTAIL_USER_EDIT_FORM
* WAGTAIL_USER_TIME_ZONES
* WAGTAILADMIN_NOTIFICATION_FROM_EMAIL
* WAGTAILADMIN_NOTIFICATION_INCLUDE_SUPERUSERS
* WAGTAILADMIN_NOTIFICATION_USE_HTML
* WAGTAILADMIN_PERMITTED_LANGUAGES
* WAGTAILADMIN_RECENT_EDITS_LIMIT
* WAGTAILADMIN_USER_LOGIN_FORM
* WAGTAILEMBEDS_FINDERS
* WAGTAILIMAGES_IMAGE_MODEL
* WAGTAILIMAGES_MAX_UPLOAD_SIZE
* WAGTAILSEARCH_BACKENDS
* WAGTAILSEARCH_HITS_MAX_AGE
* WAGTAILSEARCH_RESULTS_TEMPLATE
* WAGTAILSEARCH_RESULTS_TEMPLATE_AJAX
* WAGTAILUSERS_PASSWORD_ENABLED
* WAGTAILUSERS_PASSWORD_REQUIRED

### CODE REVIEW

**Comment by lb-** [comment]
@SalahAdDin  - yep that is added to the list - see this commit reference
https://github.com/wagtail/wagtail/pull/4535/files#diff-aac1a9060604935d8f47932fe87a54dfR573

**Comment by SalahAdDin** [comment]
@lb- but that setting still works with `draftail`.

**Comment by lb-** [comment]
@SalahAdDin  - ahh I see now. I will look at getting this PR updated with that soon. I will also look at updating the page here: http://docs.wagtail.io/en/latest/advanced_topics/customisation/extending_draftail.html

Reference docs: https://github.com/springload/wagtaildraftail#configuration

@gasman  + @loicteixeira - anything else I should include that is not listed on the wagtaildraftail repo's readme.

**Comment by loicteixeira** [comment]
@lb- I can't think of anything else right now. That's some great work on which we can iterate later if we've missed any. I think this can be merged :+1:

File: docs/advanced_topics/settings.rst [hunk 1]
```diff
@@ -250,15 +250,20 @@ Images
...
 
 This setting lets you override the maximum upload size for images (in bytes). If omitted, Wagtail will fall back to using its 10MB default value.
 
+.. code-block:: python
+
+    WAGTAILIMAGES_FEATURE_DETECTION_ENABLED = True
+
+This setting enables feature detection once OpenCV is installed, see all details on the :ref:`image_feature_detection` documentation.
+
+The ``WAGTAILIMAGES_BACKENDS`` setting is no longer supported, it was introduced in 0.3.
```
Comment (gasman) [inline docs/advanced_topics/settings.rst]: I don't think there's much point in mentioning `WAGTAILIMAGES_BACKENDS` - it was dropped in Wagtail 1.0, which is ancient history now :-)

**Comment by gasman** [comment]
Have added full documentation for `WAGTAILADMIN_RICH_TEXT_EDITORS` in 5c12356030fd7e0770e64f435966297b9b7895c2 - please review and merge if happy :-)

I don't think there's anything else from the wagtaildraftail package that needs to go in here - the Draftail integration in Wagtail 2.x is a different implementation to wagtaildraftail, and anything beyond setting a `features` list is in the realm of the "extending Draftail" docs.

**Comment by lb-** [comment]
Fixed & Merged in https://github.com/wagtail/wagtail/commit/b2e139bbd39ab2eb1923eefa632a615f0bc7abaa

**Pull request closed without merging**